Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -0,0 +1,217 @@
+                return peer != null ? this.ProviderFromPeer(peer) : null;
+            }
+        }
+
+        private CarouselItem OwnerCarouselItem
+        {
+            get { return this.Owner as CarouselItem; }
+        }
+
+        /// <summary>Adds the current element to the collection of selected items.</summary>
+        public void AddToSelection()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent?.SetSelectedItem(owner);
+        }
+
+        /// <summary>Removes the current element from the collection of selected items.</summary>
+        public void RemoveFromSelection()
+        {
+            // Cannot remove the selection of a Carousel control.
+        }
+
+        /// <summary>Clears any existing selection and then selects the current element.</summary>
+        public void Select()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent?.SetSelectedItem(owner);
+        }
+
+        /// <summary>
+        /// Gets the control type for the element that is associated with the UI Automation peer.
+        /// </summary>
+        /// <returns>The control type.</returns>
+        protected override AutomationControlType GetAutomationControlTypeCore()
+        {
+            return AutomationControlType.ListItem;
+        }
+
+        /// <summary>
+        /// Called by GetClassName that gets a human readable name that, in addition to AutomationControlType,
+        /// differentiates the control represented by this AutomationPeer.
+        /// </summary>
+        /// <returns>The string that contains the name.</returns>
+        protected override string GetClassNameCore()
+        {
+            return Owner.GetType().Name;
+        }
+
+        /// <summary>
+        /// Called by GetName.
+        /// </summary>
+        /// <returns>
+        /// Returns the first of these that is not null or empty:
+        /// - Value returned by the base implementation
+        /// - Name of the owning CarouselItem
+        /// - Carousel class name
+        /// </returns>
+        protected override string GetNameCore()
+        {
+            string name = AutomationProperties.GetName(this.OwnerCarouselItem);
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            name = this.OwnerCarouselItem.Name;
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            var textBlock = this.OwnerCarouselItem.FindDescendant<TextBlock>();
+            if (textBlock != null)
+            {
+                return textBlock.Text;
+            }
+
+            name = base.GetNameCore();
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
+
+            return string.Empty;
```

Comment:
Why exactly are you doing this? I doubt that the base peer would return null. I think just returning `base.GetNameCore()` is fine here, a lot of WinUI controls also do that.

Reply:
Very true. This was most likely in my daze of playing around with the different outputs to see what this might look like for this.